### PR TITLE
[groheondus] Export the waterconsumption value of the grohe sense guard

### LINKED ISSUE #4554: [groheondus] START_ARRAY issue
Hi,

Im getting constantly an error on a sense (having 2 plus a guard).

In the log I see on startup:
changed from ONLINE to OFFLINE (COMMUNICATION_ERROR): Cannot deserialize instance of `org.grohe.ondus.api.model.BaseAppliance` out of START_ARRAY token

It looks like  a random issue on what sense this occurs.

### PR DESCRIPTION
**Pull request by FlorianSW**
The waterconsumption is a combined metric of the total consumed water
registered by the grohe sense guard. The channel allows you to get the
value of a specific timeframe (in days) from 1 to up to 90 days (the 90
days is a restriction of the api).

This also fixes (by an update of the api library) the issue of
deserializing an array to an actual appliance.

Fixes #4554

Signed-off-by: Florian <florian.schmidt.welzow@t-online.de>

### CODE REVIEW

File: addons/binding/org.openhab.binding.groheondus/ESH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -124,10 +125,22 @@
 		<description>The humidity reported by the device</description>
 		<state readOnly="true" pattern="%.1f %unit%" />
 	</channel-type>
-	<channel-type id="temperature">
-		<item-type>Number:Temperature</item-type>
-		<label>Temperature</label>
-		<description>The temperature reported by the device</description>
-		<state readOnly="true" pattern="%.1f %unit%" />
-	</channel-type>
+    <channel-type id="temperature">
```
[the remark this answers is not in the document]
Comment (FlorianSW) [reply addons/binding/org.openhab.binding.groheondus/ESH-INF/thing/thing-types.xml]: uhhh.. sure :)

**Comment by kaikreuzer** [comment]
Did you see that compilation fails with
```
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:1.2.0:compile (default-compile) on project org.openhab.binding.groheondus: Compilation failure: Compilation failure: 
[ERROR] /home/travis/build/openhab/openhab2-addons/addons/binding/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/discovery/GroheOndusDiscoveryService.java: 
[ERROR] 	/**
[ERROR] 	^
[ERROR] Internal compiler error: java.util.ServiceConfigurationError: javax.annotation.processing.Processor: Provider lombok.launch.AnnotationProcessorHider$AnnotationProcessor could not be instantiated at java.util.ServiceLoader.fail(ServiceLoader.java:232)
[ERROR] 1 problem (1 error)
```
I actually do not really know what that means, but do you also have it locally?

**Comment by FlorianSW** [comment]
Oh, haven't seen that (didn't built with maven, just in eclipse, though). That looks strange, I can't think of what this might mean, either :/ I'll look into it, let's see what I can find :)

**Comment by kaikreuzer** [comment]
Do you see it on a local Maven build as well?

**Comment by FlorianSW** [comment]
Yep, unfortunately, I do :( However, I didn't see them with the prior (0.0.4) library version, however, the changes between these versions should not result in such issues :/

**Comment by kaikreuzer** [comment]
The weird thing is that you didn't even touch the class where it is complaining now...

**Comment by FlorianSW** [comment]
I don't think that it is actually related to the source code of the binding itself, at least I, locally, get the error at another file:

```
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:1.2.0:compile (default-compile) on project org.openhab.binding.groheondus: Compilation failure: Compilation failure:
[ERROR] .\openhab2-addons\addons\binding\org.openhab.binding.groheondus\src\main\java\org\openhab\binding\groheondus\internal\handler\GroheOndusSenseGuardHandler.java:
[ERROR] /**
[ERROR] ^
[ERROR] Internal compiler error: java.util.ServiceConfigurationError: javax.annotation.processing.Processor: Provider lombok.launch.AnnotationProcessorHider$AnnotationProcessor could not be instantiated at java.util.ServiceLoader.fail(ServiceLoader.java:232)
[ERROR] 1 problem (1 error)
[ERROR] -> [Help 1]
```

**Comment by wborn** [comment]
It looks like some black magic [Lombok](https://projectlombok.org/) processing gone wrong caused by the Lombok inside the ondus-api-0.0.6.jar.

**Comment by FlorianSW** [comment]
Yeah, but hte question is "what" :/ The version of lombok is the same in 0.0.6 and 0.0.4 and the build of the library (with gradle) works fine :(

**Comment by wborn** [comment]
I'm not into Lombok. But it appears in the new jar the `lombok/launch/Main.class` no longer exists:

meld

**Pull request merged**